### ISSUE TITLE
Country code when invoicing outside Switzerland

### ISSUE THREAD

**Issue by galchevita** [manuelbl/SwissQRBill#63]
Hello,
First of all I'd like to thank you for the great work you did!

I'd like to propose an improvement to be done in connection to the latest version of the implementation guide: [Swiss Implementation Guidelines for the QRbill (Version 2.2)](https://www.paymentstandards.ch/dam/downloads/ig-qr-bill-en.pdf) , also linked to #62 .
In the document it is written: "_For **invoicing to** countries outside Switzerland, including Lichtenstein, the country code should be printed on the payment section._ " (p.18/19 Comments column for _Account / Payable to_ heading) and my understanding is that the decision to display the country code has to take into account the country of the debtor for both creditor and debtor's address, i.e. even if the creditor is with country code CH, it has to be displayed in cases when the debtor is outside Switzerland (and LI too).

I'd like to know your thoughts about this topic.

**Comment by manuelbl** [comment]
Thanks for input.

I've never read the specification like that. But it does indeed make sense.

I'll take care of it.

**Comment by robinmamie** [comment]
> [...] my understanding is that the decision to display the country code has to take into account the country of the debtor for both creditor and debtor's address, i.e. even if the creditor is with country code CH, it has to be displayed in cases when the debtor is outside Switzerland (and LI too).

I agree that the text can be interpreted that way, but the examples I've linked in #62 do not show this behaviour (figure 18-19, p. 51-52).

**Comment by manuelbl** [comment]
As the specification is a bit ambiguous, I'm trying to understand the options and motivations. The table below shows all four combinations, and the discussion is about case 3.

| Case | Creditor address | Debtor address | Country code for creditor | Country code for debtor |
| -- | -- | -- | -- | -- |
| 1 | Switzerland | Switzerland | No | No |
| 2 | Abroad | Switzerland | Yes | No |
| 3 | Switzerland | Abroad | **Yes**| Yes |
| 4 | Abroad | Abroad | Yes | Yes |

@galchevita What would be the motivation to show the creditors country code even though it's "CH"? I have an idea but would like to hear your opinion.

**Comment by galchevita** [comment]
Hello, 
my motivation is that even the QR Bill is a Swiss payment standard, for a customers outside Switzerland it is more intelligible to have the country written on the paper instead of guessing it.
Indeed it is my personal opinion and I really appreciate your willingness to handle this topic.

**Comment by manuelbl** [comment]
Version 3.0.4 implements this, including CH country code for creditor if payer is abroad.

**Comment by galchevita** [comment]
Thank you very much!
